src/routes/search/index.js

```diff
@@ -1,517 +1,854 @@
+		return e => {
+			e.preventDefault();
+			this.setState({ currentLocation: null });
+			this.setState({ showdisqus: false });
+		};
+	}
+	hide() {
+		this.setState({ visible: false });
+		this.setState({ currentLocation: null });
+	}
+	hidesuggest() {
+		this.setState({ visiblesuggest: false });
+	}
+	hidedeletion() {
+		this.setState({ visibledelete: false });
+		this.setState({ currentLocation: null });
+	}
+	handleChange(event) {
+		this.setState({ value: event.target.value });
+	}
+	handleSubmit(event) {
+		this.setState({ loading: false });
+		mixpanel.track("User searched for: " + this.state.value);
+		axios
+			.get(
+				"https://api.govote.org.ng/search?query=" +
+					this.state.value +
+					"&key=k9ihbvse57fvsujbsvsi5362WE$NFD2"
+			)
+			.then(res => {
+				this.setState({ loading: true });
+				this.setState({ locations: res.data.data });
+				this.setState({ result: true });
+				route(`${this.props.path}?query=${this.state.value}`);
+			})
+			.catch(err => {
+				console.error(err);
+			});
+		event.preventDefault();
+	}
 
-  constructor () {
-    super()
-    this.state = {
-      result: false,
-      value: '',
-      loading: false,
-      locations: [],
-      visible: false,
-      visiblesuggest: false,
-      visibledelete: false,
-      showdisqus: false,
-      currentLocation: null
-    }
+	constructor() {
+		super();
+		this.state = {
+			result: false,
+			value: "",
+			loading: false,
+			locations: [],
+			visible: false,
+			visiblesuggest: false,
+			visibledelete: false,
+			showdisqus: false,
+			currentLocation: null
+		};
 
-    this.handleChange = this.handleChange.bind(this)
-    this.handleSubmit = this.handleSubmit.bind(this)
-    this.show = this.show.bind(this)
-    this.showdeletion = this.showdeletion.bind(this)
-    this.showsuggest = this.showsuggest.bind(this)
-    this.hide = this.hide.bind(this)
-    this.hidedeletion = this.hidedeletion.bind(this)
-    this.hidesuggest = this.hidesuggest.bind(this)
-    this.showDisqus = this.showDisqus.bind(this)
-    this.closeDisqus = this.closeDisqus.bind(this)
-  }
+		this.handleChange = this.handleChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.show = this.show.bind(this);
+		this.showdeletion = this.showdeletion.bind(this);
+		this.showsuggest = this.showsuggest.bind(this);
+		this.hide = this.hide.bind(this);
+		this.hidedeletion = this.hidedeletion.bind(this);
+		this.hidesuggest = this.hidesuggest.bind(this);
+		this.showDisqus = this.showDisqus.bind(this);
+		this.closeDisqus = this.closeDisqus.bind(this);
+	}
+	componentWillMount() {
+		let { query } = this.props.matches;
+		query = !query ? " " : query;
+		const sessionStorageRef = JSON.parse(
+			sessionStorage.getItem(`PVCLocations`)
+		);
+		// check if session storage already exists, if not make a get request to the API
+		if (sessionStorageRef) {
+			this.setState({
+				value: query,
+				loading: true,
+				locations: sessionStorageRef
+			});
```

Comment:
You can [return early](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) here...